hubblestack/extmods/fdg/grep.py

```diff
@@ -14,13 +14,42 @@
 log = logging.getLogger(__name__)
 
 
-def grep(path, pattern, grep_args=None, format_chained=True, chained=None):
+def file(path, pattern, grep_args=None, format_pattern=True, format_chained=True, chained=None):
     '''
     Given a target ``path``, call ``grep`` to search for for ``pattern`` in that
     file.
 
     By default, the ``pattern`` will have ``.format()`` called on it with
     ``chained`` as the only argument. (So, use ``{0}`` in your pattern to
+    substitute the chained value.)
+    Set ``format_pattern`` to Falso to format the ``path`` instead.
+    If you want to avoid having to escape curly braces, set ``format_chained=False``.
+
+    The first return value (status) will be True if the pattern is found, and
+    False othewise. The second argument will be the output of the ``grep``
+    command.
+
+    ``grep_args`` can be used to pass in arguments to grep.
+    '''
+    if format_chained:
+        if format_pattern:
+            pattern = pattern.format(chained)
+        else:
+            path = path.format(chained)
+    if grep_args is None:
+        grep_args = []
+    ret = _grep(pattern=pattern, path=path, *grep_args)
+    status = bool(ret)
+    return status, ret
+
+
+def stdin(pattern, starting_string=None, grep_args=None, format_chained=True, chained=None):
+    '''
+    Given a target string, call ``grep`` to search for for ``pattern`` in that
+    string.
+
+    By default, the ``starting_string`` will have ``.format()`` called on it with
+    ``chained`` as the only argument. (So, use ``{0}`` in your pattern to
     substitute the chained value.) If you want to avoid having to escape
```

Comment:
Can you add a note here that if you don't provide a starting string, the chained value will just be used? I think it's mostly common sense but I wanted to explicitly call it out.